src/frontend/src/binder/insert.rs

```diff
@@ -48,6 +52,11 @@ impl Binder {
             .map(|c| c.data_type.clone())
             .collect();
 
+        let vnode_mapping = self
+            .bind_table(&schema_name, &table_name, None)?
+            .table_catalog
+            .vnode_mapping;
```

Comment:
Binding tables may bring some side effects like adding columns to the context. I'm not sure whether this is correct, if we write something like `insert into t select * from t`.

BTW, it seems like this is much more likely to be a property of the table **source**? Can we record it in the source catalog?

Reply:
Yes, you're right. I fixed this by avoid calling bind directly. For two, the `vnode_mapping` will be moved in next PR, so this won't be exist in source nor table catalog.